keras_cv/layers/preprocessing/mix_up.py

```diff
@@ -0,0 +1,90 @@
+import tensorflow as tf
+import tensorflow.keras as keras
+import tensorflow.keras.layers as layers
+import warnings
+
+
+class MixUp(layers.Layer):
+    """
+    MixUp implements the MixUp data augmentation technique as proposed in https://arxiv.org/abs/1710.09412.
+
+    Args:
+        alpha: alpha parameter for the sample distribution.  Defaults 0.8.
+        probability: probability to apply the MixUp augmentation.  Default 1.0.
+        label_smoothing: coefficient used in label smoothing.  Default 0.0.
+
+    Sample usage:
+    ```python
+    (images, labels), _ = tf.keras.datasets.cifar10.load_data()
+    mixup = keras_cv.layers.preprocessing.mix_up.MixUp(10)
+    augmented_images, updated_labels = mixup(images, labels)
+    ```
+    """
+
+    def __init__(
+        self, alpha=0.8, probability=1.0, label_smoothing=0.0, seed=None, **kwargs
```

Comment:
Same comments here about argument names and descriptions

Reply:
To clarify: probability applies on a batchwise basis, not an imagewise basis.

Reply:
@fchollet perhaps `mixup_probability`?  Or is that implied by the layer name?

Reply:
the mixup/mixcut is already contained in the class name, so we can skip them in the arg name, same as we do the the dropout layer.

Reply:
Done, I just changed to rate and made it required.  I like this API.